### PR TITLE
[fuzz] fix h1 end to end fuzz test issue

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Commit Message: Removes temporary files used for direct responses in h1 and h2 end to end fuzz test.

Additional Description: This was causing a flakey issue where OSS-Fuzz's temporary path used for direct responses looked illegal to Envoy. It seems unlikely that it was one of the deny listed paths in `illegalPath` because the path is always `/mnt`. It may be flakey on a `realpath` failure when getting the canonical path. Replace with inline bytes since it's better not to mess around with the filesystem anyway.

```
[2021-02-17 14:24:17.258][49492][critical][assert] [test/integration/base_integration_test.cc:322] assert failure: rejected_counter == nullptr \|\| rejected_counter->value() == 0. Details: Lds update failed. Details
failed_configuration:
[...]
details: "Invalid path: /mnt/scratch0/clusterfuzz/bot/tmp/test_envoy"
```

and bootstrap path:
```
[2021-02-17 20:55:28.894][82532][critical][main] [source/server/server.cc:109] error initializing configuration '/mnt/scratch0/clusterfuzz/bot/tmp/bootstrap.pb': Invalid path: /mnt/scratch0/clusterfuzz/bot/tmp/bootstrap.pb'
```

Risk Level: Low
Fixes: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=31617

### CODE REVIEW

**Comment by asraa** [comment]
Just found a very similar issue with h2_capture that I will add for this PR. Will ping when ready for review.

**Comment by asraa** [comment]
Ready, sorry about that! Would love help for a better solution btw.

File: source/common/filesystem/posix/filesystem_impl.cc [hunk 1]
```diff
@@ -160,6 +160,14 @@ bool InstanceImplPosix::illegalPath(const std::string& path) {
...
     return true;
   }
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+  // Allow temporary files in OSS-Fuzz to prevent failures running (h1/h2)_capture_fuzz_test.
+  // Ideally this would be TestEnvironment::temporaryPath() but this avoids depending on test
+  // libraries.
+  if (absl::StartsWith(canonical_path.rc_, "/mnt")) {
+    return true;
+  }
+#endif
```
Comment (adisuissa) [inline source/common/filesystem/posix/filesystem_impl.cc:170]: Does this mean that the fuzzer code won't be able to access files in _/mnt_?
Could this impact fuzz tests that may need to access external files under the _/mnt_ directory?

**Review by adisuissa — COMMENTED** [review]
Strange bug...
I'm guessing that because the test was able to write the file upon initialization without a failure, and because it only failed on the 33rd iteration (although the previous may have been wrong input) then it might be something external that deletes temporary files, but that's just a guess.
I wonder if using _/tmp_ is the right solution (https://google.github.io/oss-fuzz/further-reading/fuzzer-environment/#file-system).

Seems that the fix for the h1 and h2 fuzzer is inconsistent (h1 uses inlined bytes, and h2 uses illegal path). Should it be that way?

**Comment by asraa** [comment]
> I wonder if using /tmp is the right solution (https://google.github.io/oss-fuzz/further-reading/fuzzer-environment/#file-system).

The best solution is probably to make `temporaryPath()` return `/tmp` for OSS-Fuzz. As long as the realpath of /tmp looks legal, things will work.

> Seems that the fix for the h1 and h2 fuzzer is inconsistent (h1 uses inlined bytes, and h2 uses illegal path). Should it be that way?

Ideally I want to avoid using the filesystem anyway.

File: source/common/filesystem/posix/filesystem_impl.cc (hunk 1 above)
Comment (asraa) [reply source/common/filesystem/posix/filesystem_impl.cc:170]: OMG thanks for pointing that out. The fix was meant to do the opposite.

**Review by adisuissa — APPROVED** [review]
Thanks for fixing this!

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**